packages/backend/server/src/plugins/copilot/context/session.ts

```diff
@@ -30,12 +60,43 @@ export class ContextSession implements AsyncDisposable {
     ) as ContextList;
   }
 
-  async addDocRecord(docId: string): Promise<ContextList> {
-    if (!this.config.docs.some(f => f.id === docId)) {
-      this.config.docs.push({ id: docId, createdAt: Date.now() });
-      await this.save();
+  private readStream(
+    readable: Readable,
+    maxSize = 50 * OneMB
+  ): Promise<Buffer<ArrayBuffer>> {
+    return new Promise<Buffer<ArrayBuffer>>((resolve, reject) => {
+      const chunks: Uint8Array[] = [];
+      let totalSize = 0;
+
+      readable.on('data', chunk => {
+        totalSize += chunk.length;
+        if (totalSize > maxSize) {
+          reject(new BlobQuotaExceeded());
```

Comment:
can we get the file size in the Request headers and reject earlier?

Reply:
added